Whiskey/Tasks/Invoke-WhiskeyNpmLicenseChecker.ps1

```diff
@@ -0,0 +1,180 @@
+
+    # Examples
+
+    ## Example 1
+
+        BuildTasks:
+        - NpmLicenseChecker:
+            NpmRegistryUri: "http://registry.npmjs.org"
+    
+    This example will run `license-checker` against the modules listed in the `package.json` file located in the build root.
+
+    ## Example 2
+
+        BuildTasks:
+        - NpmLicenseChecker:
+            NpmRegistryUri: "http://registry.npmjs.org"
+            WorkingDirectory: app
+    
+    This example will run `license-checker` against the modules listed in the `package.json` file that is located in the `(BUILD_ROOT)\app` directory.
+    #>
+
+    [Whiskey.Task("NpmLicenseChecker", SupportsClean=$true, SupportsInitialize=$true)]
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory=$true)]
+        [object]
+        $TaskContext,
+
+        [Parameter(Mandatory=$true)]
+        [hashtable]
+        $TaskParameter
+    )
+
+    Set-StrictMode -Version 'Latest'
+    Use-CallerPreference -Cmdlet $PSCmdlet -SessionState $ExecutionContext.SessionState
+
+    $startedAt = Get-Date
+    function Write-Timing
+    {
+        param(
+            $Message
+        )
+
+        $now = Get-Date
+        Write-Debug -Message ('[{0}]  [{1}]  {2}' -f $now,($now - $startedAt),$Message)
+    }
+
+    if ($TaskContext.ShouldClean())
+    {
+        Write-Timing -Message 'Cleaning'
+        $nodeModulesPath = (Join-Path -path $TaskContext.BuildRoot -ChildPath 'node_modules')
+
+        if (Test-Path $nodeModulesPath -PathType Container)
+        {
+            $outputDirectory = Join-Path -path $TaskContext.BuildRoot -ChildPath '.output' 
+            $emptyDir = New-Item -Name 'TempEmptyDir' -Path $outputDirectory -ItemType 'Directory'
+
+            Write-Timing -Message ('Emptying {0}' -f $nodeModulesPath)
+            Invoke-WhiskeyRobocopy -Source $emptyDir -Destination $nodeModulesPath | Write-Debug
+            Write-Timing -Message ('COMPLETE')
+
+            Remove-Item -Path $emptyDir
+            Remove-Item -Path $nodeModulesPath
+        }
+        return
+    }
+
+    $npmRegistryUri = $TaskParameter['NpmRegistryUri']
+    if (-not $npmRegistryUri) 
+    {
+        Stop-WhiskeyTask -TaskContext $TaskContext -Message 'Property ''NpmRegistryUri'' is mandatory. It should be the URI to the registry from which Node.js packages should be downloaded, e.g.,
+        
+        BuildTasks:
+        - NpmLicenseChecker:
+            NpmRegistryUri: https://registry.npmjs.org/
+
+        '
+    }
+
+    $workingDirectory = $TaskContext.BuildRoot
+    if ($TaskParameter['WorkingDirectory'])
+    {
+        $workingDirectory = $TaskParameter['WorkingDirectory'] | Resolve-WhiskeyTaskPath -TaskContext $TaskContext -PropertyName 'WorkingDirectory'
+    }
+
+    Push-Location -Path $workingDirectory
+    try
+    {
+        $noColorArg = ''

```

Comment:
Whether or not to provide the `--no-color` arg should be handled by `Invoke-WhiskeyNpmCommand`, I think.

Reply:
Ah yes, this can easily be moved into `Invoke-WhiskeyNpmCommand` since it already takes a `ForDeveloper` switch which can be used to determine whether or not `--no-color` is used.